### ISSUE TITLE
BottomSheetController - random crashes 

### ISSUE THREAD

**Issue by vjard33** [thomsmed/ios-examples#33]
Hello @thomsmed, Thanks for your repository, very helpful! 
I can see one crash reported from Crashlytics with this code.

```
    func start(moving presentedView: UIView, interactiveDismissal: Bool) {
        self.interactiveDismissal = interactiveDismissal

        heightAnimator?.stopAnimation(false)
        heightAnimator?.finishAnimation(at: .start)
        offsetAnimator?.stopAnimation(false)
        offsetAnimator?.finishAnimation(at: .start)

        heightAnimator = createHeightAnimator(
            animating: presentedView, from: presentedView.frame.height
        )

        if !interactiveDismissal {
            offsetAnimator = createOffsetAnimator(
                animating: presentedView, to: stretchOffset
            )
        }
    }
```

It crashes on this line:
```
        heightAnimator = createHeightAnimator(
            animating: presentedView, from: presentedView.frame.height
        )
```
This is not really frequent, but sometimes it happens with this fatal error:
```
Fatal Exception: NSGenericException
*** Collection <__NSArrayM: 0x28073bed0> was mutated while being enumerated.
```

Here is the stacktrace from Crashlytics:
<img width="1335" alt="Capture d’écran 2023-04-05 à 09 44 21" src="https://user-images.githubusercontent.com/99725306/230015063-807132fa-a069-4722-b660-c0cc1ddcc9d6.png">

Any idea how to fix this crash? Thank you so much!!

**Comment by vjard33** [comment]
Any idea @thomsmed? 🙏🤞

**Comment by thomsmed** [comment]
Hi! And sorry for my super duper late reply 😅 I had a quick look at it, and I can't see any immediate reason for this to happen (also tried to provoke a crash, but no success). But I suspect there are some edge cases where the two UIViewPropertyAnimators "steps on each others toes" (maybe if the user fiddles too much with the bottom sheet?)

Anyway. I've wanted for a long time to improve this example code, as it has some issues... And are currently working on some updates inspired from the latest example project added to this repo ([Interactive animations with UIViewPropertyAnimator](https://github.com/thomsmed/ios-examples/tree/main/InteractiveAnimations)).

Hopefully I'll have an updated BottomSheetController example ready soon! And I hope that will also address weird issues like the one you are experiencing.

It makes me really happy that people are actually finding these example project useful, so keep the feedback coming! 😄 And I'll try to reply faster in the future 😅

**Comment by vjard33** [comment]
Thank you so much, cannot wait for that!! 🙏

**Comment by thomsmed** [comment]
Alright! I have now updated the example code for BottomSheetController. I think the changes bring great improvements to both code and behaviour of the bottom sheet.

Merged with #37 

Go and check it out! 😄 It depends a little bit on how you've utilised this example code in your own app, but everything should work as before 🙏

Hopefully these changes will also address the issues you were seeing.

Closing this GitHub issue.
